### ISSUE TITLE
Device Support

### ISSUE THREAD

**Issue by ritdaw** [MxLabs/Anviz#15]
Which Anviz devices are supported? Maybe an update to the readme would be good :)

**Comment by MxLabs** [comment]
We have tested on W1 Pro and WF x series. If you have a different device, we can add it on readme.

**Comment by ritdaw** [comment]
Great thanks! Haven't got one yet but obviously the library working with a specific device helps us make a decision.

**Comment by JohnKiller** [comment]
I have the W1 and also the W2 should work since it's basically the same hardware plus the relay to open the door.

**Comment by scorredoira** [comment]
I have an A380 and it works too.

**Comment by JohnKiller** [comment]
Let's keep this issue open since I'm currently implementing more features in the next week. I would like some community feedback for the fingerprint templates and users edit once I'm done with them. Another thing that I'm currently working on is the server mode.

**Comment by ritdaw** [comment]
Thanks all, amazing to see such a quick response!

**Comment by ritdaw** [comment]
Out of interest, what is the main difference between the W1 & the W1 Pro? Will both work?

**Comment by elund** [comment]
I have tested with an Anviz M3 device, and it does not work out of the box with this .NET library. It looks like that the sample program can connect to the M3 in client mode, but GetDeviceID() never returns with the id. The official SDK demo from Anviz works fine with M3.

**Comment by Pretzel911** [comment]
I'm using it with an A300, seems to mostly work, except GetDeviceID(), which returns a set of numbers that are unrelated to the Device ID, and seems to crash the my program the next time data is requested from the device.

**Comment by elund** [comment]
I have tested a bit more with an Anviz M3 device and I cannot get it to work with this .NET library, so I have to stick to the official SDK from Anviz.

**Comment by aljohn-ma** [comment]
Doesn't work on my Anviz EP300.

**Comment by fgbustoa** [comment]
Doesn't work on A300. It seems to connect, but any Get command stuck.

**Comment by aljohn-ma** [comment]
The issue is on the device id. SDK gives different device id that's why it gets "stuck" when trying to access other functions. 

Try using the device id  the SDK provided on your device.

**Comment by noelching** [comment]
Not working in EP300

**Comment by MicaelMor** [comment]
I too was unable to have it work by default with my A300, however as aljohn-ma mentioned this is a device ID issue, so after digging it around I have found the change required to fix this issue for the A300 devices, and likely EP300 since I have had one in the past and it seemed to be very close to the A300.

I have created an issue for this specific thing #40

**Comment by JohnKiller** [comment]
Hi everyone,
I've just pushed to NuGet version 2.0.9 to address @MicaelMor problems.
This may enable more device compatibility. Please report back if you got some new device working.
Thanks

**Comment by jinehme** [comment]
Hi. I have read the comments and I know John mentioned that the W2 should work, but has anyone tested it? 
Thanks

**Comment by iboras** [comment]
Hi
Did you try/implement Anviz EP300?

**Comment by jlewis0** [comment]
Hi, i wanna know if is possible to send sound or images to screen on my w1 pro

**Comment by epastinante** [comment]
> Hi. I have read the comments and I know John mentioned that the W2 should work, but has anyone tested it? Thanks

Yes, it works. But everytime you got an error response from the device, the connection close. For example, enrollment fail, trying to erase an user.

I notice when SDK raise an error normally hangs the communication with the device.

Other thing is can't handle ping and pong msg. This events are never triggered. DeviceError yes


I'm using VB.net with 4.7.2 .net framework.
